### PR TITLE
Fixed#5400: Added Kafka Event Publisher

### PR DESCRIPTION
**Pull request by parthp2107**
### Describe your changes :

Fixes: #5400

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@mohitdeuex

### CODE REVIEW

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/kafka/KafkaWebhookEventPublisher.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    properties.put(ProducerConfig.LINGER_MS_CONFIG, webhook.getKafkaProperties().getLingerMS());
+    properties.put(ProducerConfig.BUFFER_MEMORY_CONFIG, webhook.getKafkaProperties().getBufferMemory());
+    properties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, webhook.getKafkaProperties().getKeySerializer());
+    properties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, webhook.getKafkaProperties().getValueSerializer());
+    return properties;
+  }
+
+  @Override
+  public void publish(EventResource.ChangeEventList events) throws JsonProcessingException {
+    producer = new KafkaProducer<>(setKafkaProperties(webhook));
```
Comment (mohitdeuex) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/kafka/KafkaWebhookEventPublisher.java:53]: again we are creating here , we don't need to create the producer on every change event

**Review by mohitdeuex — CHANGES_REQUESTED** [review]

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/kafka/KafkaWebhookEventPublisher.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  public void publish(EventResource.ChangeEventList events) throws JsonProcessingException {
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, webhook.getEndpoint().toString());
+    props.put(ProducerConfig.ACKS_CONFIG, webhook.getKafkaProperties().getAcks());
+    props.put(ProducerConfig.RETRIES_CONFIG, webhook.getKafkaProperties().getRetries());
+    props.put(ProducerConfig.BATCH_SIZE_CONFIG, webhook.getBatchSize());
+    props.put(ProducerConfig.LINGER_MS_CONFIG, webhook.getKafkaProperties().getLingerMS());
+    props.put(ProducerConfig.BUFFER_MEMORY_CONFIG, webhook.getKafkaProperties().getBufferMemory());
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, webhook.getKafkaProperties().getKeySerializer());
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, webhook.getKafkaProperties().getValueSerializer());
+    producer = new KafkaProducer<>(props);
```
[the remark this answers is not in the document]
Comment (parthp2107) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/kafka/KafkaWebhookEventPublisher.java]: Addressed in the latest commit.

File: catalog-rest-service/src/main/resources/json/schema/api/events/createWebhook.json [hunk 3]
```diff
@@ -51,17 +51,24 @@
...
+    "kafkaProperties": {
+      "description": "Properties of Kafka Producer",
+      "$ref": "../../configuration/kafkaEventConfiguration.json"
+    },
     "webhookType": {
       "description": "Type of webhook slack,generic etc",
       "type": "string",
       "default": "generic",
-      "enum": ["slack", "generic"],
+      "enum": ["slack", "generic", "kafka"],
```
[the remark this answers is not in the document]
Comment (parthp2107) [reply catalog-rest-service/src/main/resources/json/schema/api/events/createWebhook.json:62]: Addressed in the latest commit.

File: conf/openmetadata.yaml [hunk 4]
```diff
@@ -220,6 +220,15 @@ slackEventPublishers:
         entities:
          - "*"
 
+kafkaEventPublishers:
```
[the remark this answers is not in the document]
Comment (parthp2107) [reply conf/openmetadata.yaml]: Addressed in the latest commit.

File: catalog-rest-service/src/main/resources/json/schema/configuration/kafkaEventConfiguration.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    "keySerializer": {
+      "description": "Serializer class for key",
+      "type": "string"
+    },
+    "valueSerializer": {
+      "description": "Serializer class for value",
+      "type": "string"
+    }
+  },
+  "required": ["topics", "keySerializer", "valueSerializer"],
```
Comment (mohitdeuex) [inline catalog-rest-service/src/main/resources/json/schema/configuration/kafkaEventConfiguration.json:42]: topic is only required , for other two serialised we can add default value of *.StringSerializer that we had saw earlier

Comment (mohitdeuex) [reply catalog-rest-service/src/main/resources/json/schema/configuration/kafkaEventConfiguration.json:42]: we can have opinionated instantiation , with default value for most of the properties , in json give default properties rest of the things that are mandatory like topic should be added to required

**Review by mohitdeuex — CHANGES_REQUESTED** [review]

File: catalog-rest-service/src/main/resources/json/schema/configuration/kafkaEventConfiguration.json (hunk 5 above)
Comment (parthp2107) [reply catalog-rest-service/src/main/resources/json/schema/configuration/kafkaEventConfiguration.json]: Addressed in the latest commit.

**Review by mohitdeuex — APPROVED** [review]

**Pull request merged**